gpu_bdb/bdb_tools/text.py

```diff
@@ -16,24 +16,36 @@
 
 import os
 
-
 EOL_CHAR = "è"
 
 
 def create_sentences_from_reviews(
-    df, review_column="pr_review_content", end_of_line_char=EOL_CHAR
+    df, review_column="pr_review_content", end_of_line_char=EOL_CHAR,
 ):
+    import pandas as pd
     import cudf
-
-    sentences = df[review_column].str.tokenize(delimiter=end_of_line_char)
-
-    # expand the reviews
-    tk_cnts = df[review_column].str.token_count(delimiter=end_of_line_char)
-
+    import numpy as np
+    
+    if isinstance(df, cudf.DataFrame):
+        sentences = df[review_column].str.tokenize(delimiter=end_of_line_char)   
+        tk_cnts = df[review_column].str.token_count(delimiter=end_of_line_char)
+    else:
+        sentences = df[review_column].str.split(end_of_line_char)
+        tk_cnts = sentences.str.len()
+        sentences = sentences.explode(ignore_index=True)
+        
+        
     # use pr_review_sk as the global position
     # (leaving hardcoded as it's consistent across all queries)
     global_pos = df.pr_review_sk.repeat(tk_cnts).reset_index(drop=True)
-    out = cudf.DataFrame({"sentence": sentences, "review_idx_global_pos": global_pos})
+    
+    if isinstance(df, cudf.DataFrame):
+        out = cudf.DataFrame({"sentence": sentences, "review_idx_global_pos": global_pos})
+    else:
+        out = pd.DataFrame({"sentence": sentences, "review_idx_global_pos": global_pos})
+        out["sentence"] = out.sentence.replace('', np.nan)
+        out = out.dropna().reset_index(drop=True)
```

Comment:
Why are we only doing this in pandas and not cuDF. We should not do this kind of branching .

Reply:
The issue is that when the pandas `str.strip` functions encounters directly adjacent instances of the delimeter it adds empty strings to the output to represent those splits. But, the cudf `tokenize` functions does not. So this removes those extra empty strings. I think if we switch to using `str.strip` for both cudf and pandas we'll be able to avoid branching here.